# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/base/nsFrameLoader.h
# Commit: f95c19e45ab2
# Full Hash: f95c19e45ab2ef6a1ba8ac8d4124df117928b1a4
# Author: Jonathan Watt <jwatt@jwatt.org>
# Date: 2020-07-22 21:55:45
# Regressor Bug: 1652270
# File Overlap Count: 1
# Description:
#   Bug 1652270. Convert nsFrameLoader::Print to return a Promise. r=farre,remote-protocol-reviewers,marionette-reviewers,jgraham,whimboo,mixedpuppy
#   
#   (Instead of requiring callers to pass an nsIWebProgressListener.)
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D83249
# ==============================================================================

diff -r c749c42dc6b9 -r f95c19e45ab2 dom/base/nsFrameLoader.h
--- a/dom/base/nsFrameLoader.h	Wed Jul 22 18:40:32 2020 +0300
+++ b/dom/base/nsFrameLoader.h	Wed Jul 22 15:54:01 2020 +0000
@@ -24,6 +24,7 @@
 #include "mozilla/dom/Document.h"
 #include "mozilla/dom/Element.h"
 #include "mozilla/dom/Nullable.h"
+#include "mozilla/dom/Promise.h"
 #include "mozilla/dom/WindowProxyHolder.h"
 #include "mozilla/layers/LayersTypes.h"
 #include "nsCOMPtr.h"
@@ -105,6 +106,7 @@
   typedef mozilla::dom::BrowserBridgeChild BrowserBridgeChild;
   typedef mozilla::dom::BrowsingContext BrowsingContext;
   typedef mozilla::dom::BrowsingContextGroup BrowsingContextGroup;
+  typedef mozilla::dom::Promise Promise;
 
  public:
   // Called by Frame Elements to create a new FrameLoader.
@@ -223,9 +225,9 @@
 
   void RequestSHistoryUpdate(bool aImmediately = false);
 
-  void Print(uint64_t aOuterWindowID, nsIPrintSettings* aPrintSettings,
-             nsIWebProgressListener* aProgressListener,
-             mozilla::ErrorResult& aRv);
+  already_AddRefed<Promise> Print(uint64_t aOuterWindowID,
+                                  nsIPrintSettings* aPrintSettings,
+                                  mozilla::ErrorResult& aRv);
 
   void StartPersistence(BrowsingContext* aContext,
                         nsIWebBrowserPersistDocumentReceiver* aRecv,